src/saltext/vmware/modules/esxi.py

```diff
@@ -1166,6 +1166,246 @@ def get_user(
         raise salt.exceptions.SaltException(str(exc))
 
 
+def add_role(
+    role_name,
+    privilege_ids,
+    esxi_host_name=None,
+    esxi_user_name=None,
+    esxi_user_password=None,
+    service_instance=None,
+):
+    """
+    Add local role on ESXi host.
+
+    role_name
+        Role to create on ESXi host. (required).
+
+    privilege_ids
+        List of privileges for the role. (required).
+        Refer: https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.security.doc/GUID-ED56F3C4-77D0-49E3-88B6-B99B8B437B62.html
```

Comment:
I love the link to documentation right here. 

Looking at the page, I'm not sure what the privileges should look like. 

This is because some names are unusual, like"Alarms.Disable or enable alarm on entity". I wondered if the docs were messed up and the correct name was "Alarms.Disable", but then I also saw "Alarms.Disable alarm action".

A user can figure it out via trial and error, those docs could be improved, or if those *are* correct privileges, I think it would be nice to show them here as an example.

Reply:
That's the best document available. The API has no documentation at all for this enumeration. I can add a few examples, but that will not be exhaustive. Ansible refers to a similar documentation example.